### PR TITLE
Implemented error alerts

### PR DESCRIPTION
**Pull request by samuelerizzo**

### CODE REVIEW

File: CodingAChat/Scenes/Chat/ChatCoordinator.swift [hunk 1]
```diff
@@ -64,7 +64,11 @@ class ChatCoordinator: Coordinator {
 extension ChatCoordinator: ChatViewControllerDelegate {
     
     func sendMessage(message: String, type: Type) {
-        guard let name = user.displayName, let id = Auth.auth().currentUser?.uid else { return print("User without name") }
+        guard let name = user.displayName, let id = Auth.auth().currentUser?.uid else {
+            UIAlertController.show(title: "Unable to send messages", message:"Please set your name and your image first", from: self.chatViewController)
```
Comment (cbarbera80) [inline CodingAChat/Scenes/Chat/ChatCoordinator.swift]: please add a space after message:

File: CodingAChat/Scenes/Chat/ChatCoordinator.swift [hunk 2]
```diff
@@ -64,7 +64,11 @@ class ChatCoordinator: Coordinator {
...
+            print("User without name")
```
Comment (cbarbera80) [inline CodingAChat/Scenes/Chat/ChatCoordinator.swift]: Please remove unuseful print

File: CodingAChat/Scenes/Edit user details/EditDetailsCoordinator.swift [hunk 3]
```diff
@@ -46,7 +46,7 @@ extension EditDetailsCoordinator: EditDetailsViewControllerDelegate {
         }.ensure {
             SVProgressHUD.dismiss()
         }.catch { error in
-            //TODO: showalert
+            UIAlertController.show(message: "Error while saving data", from: self.viewController)
```
Comment (cbarbera80) [inline CodingAChat/Scenes/Edit user details/EditDetailsCoordinator.swift]: Please remove print

File: CodingAChat/Scenes/Chat/ChatCoordinator.swift (hunk 2 above)
Comment (samuelerizzo) [reply CodingAChat/Scenes/Chat/ChatCoordinator.swift]: Ok

File: CodingAChat/Scenes/Home/HomeCoordinator.swift [hunk 4]
```diff
@@ -51,6 +51,7 @@ extension HomeCoordinator: HomeViewControllerDelegate {
         db.collection("channels").addSnapshotListener() { [weak self] (querySnapshot, err) in
             
             if let err = err {
+                UIAlertController.show(message: "Unable to load channels", from: self!.homeViewController)
```
Comment (cbarbera80) [inline CodingAChat/Scenes/Home/HomeCoordinator.swift]: this self! maybe lead some crash. If you need strong self please add guard let self = self else { return }

File: CodingAChat/Scenes/Login/LoginCoordinator.swift [hunk 5]
```diff
@@ -40,6 +40,7 @@ extension LoginCoordinator: LoginViewControllerDelegate {
                 self?.onLogin?(result.user)
                 
             } else if let error = error {
+                UIAlertController.show(message: "Wrong email or password", from: self!.loginViewController)
```
Comment (cbarbera80) [inline CodingAChat/Scenes/Login/LoginCoordinator.swift]: Same here, you could add let self = self in if condition and safety remove self!

File: CodingAChat/Scenes/Chat/ChatCoordinator.swift (hunk 1 above)
Comment (samuelerizzo) [reply CodingAChat/Scenes/Chat/ChatCoordinator.swift]: ??

**Review by cbarbera80 — APPROVED** [review]

**Pull request merged**